File: select/select.exs
defmodule Select do
  @doc """
  ## buble sort
  """
  def sort([]), do: [] 

  def sort(list) do 
    a = Enum.reduce(list, hd(list), &adje/2 )
    [a | sort(list -- [a])]
  end
  
  def adje(x, acc) do 
    if x > acc do 
      acc
    else
      x
    end
  end
end

File: select/select.erl
-module(select).
-export([sort/1]).

sort([]) -> [];
sort(List) ->
  H  = switch(List),
  [H | sort(List -- [H])].

switch([]) -> [];
switch([X]) -> X;
switch([X,Y | Tail]) when X > Y ->
  switch([Y|Tail]);
switch([X,Y | Tail]) -> 
  switch([X|Tail]).


